packages/expect/src/matchers.ts

```diff
@@ -368,57 +372,84 @@ const matchers: MatchersObject = {
 
   toContain(
     this: MatcherState,
-    collection: ContainIterable | string,
-    value: unknown,
+    received: ContainIterable | string,
+    expected: unknown,
   ) {
-    const collectionType = getType(collection);
+    const isNot = this.isNot;
+    const options: MatcherHintOptions = {
+      comment: 'indexOf',
+      isNot,
+      promise: this.promise,
+    };
 
-    let converted: any = null;
-    if (Array.isArray(collection) || typeof collection === 'string') {
-      // strings have `indexOf` so we don't need to convert
-      // arrays have `indexOf` and we don't want to make a copy
-      converted = collection;
-    } else {
-      try {
-        converted = Array.from(collection);
-      } catch (e) {
-        throw new Error(
-          matcherErrorMessage(
-            matcherHint('.toContain', undefined, undefined, {
-              isNot: this.isNot,
-            }),
-            `${RECEIVED_COLOR(
-              'received',
-            )} value must not be null nor undefined`,
-            printWithType('Received', collection, printReceived),
-          ),
+    if (received == null) {
+      throw new Error(
+        matcherErrorMessage(
+          matcherHint('toContain', undefined, undefined, options),
+          `${RECEIVED_COLOR('received')} value must not be null nor undefined`,
+          printWithType('Received', received, printReceived),
+        ),
+      );
+    }
+
+    if (typeof received === 'string') {
+      const index = received.indexOf(String(expected));
+      const pass = index !== -1;
+
+      const message = () => {
+        const labelExpected = `Expected ${
+          typeof expected === 'string' ? 'substring' : 'value'
+        }`;
+        const labelReceived = 'Received string';
+        const printLabel = getLabelPrinter(labelExpected, labelReceived);
+
+        return (
+          matcherHint('toContain', undefined, undefined, options) +
+          '\n\n' +
+          `${printLabel(labelExpected)}${isNot ? 'not ' : ''}${printExpected(
+            expected,
+          )}\n` +
+          `${printLabel(labelReceived)}${isNot ? '    ' : ''}${
+            isNot
+              ? printReceivedStringContainExpectedSubstring(
+                  received,
+                  index,
+                  String(expected).length,
+                )
+              : printReceived(received)
+          }`
         );
-      }
+      };
+
+      return {message, pass};
     }
-    // At this point, we're either a string or an Array,
-    // which was converted from an array-like structure.
-    const pass = converted.indexOf(value) != -1;
+
+    const indexable = Array.isArray(received) ? received : Array.from(received);
+    const index = indexable.indexOf(expected);
```

Comment:
Can we use `includes` (also for strings)? Node 6 supports it.

Reply:
The `index` is to inverse highlight item in the report when `.not.toContainEqual` fails.

Reply:
Oh right, nevermind